spec/acceptance/init_spec.rb

```diff
@@ -42,4 +69,30 @@ class { 'bolt': use_release_package => false }
     end
     # rubocop:enable RSpec/RepeatedExampleGroupBody
   end
+
+  describe 'with ensure=absent' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        class { 'bolt':
+          version => 'absent',
+        }
+        PUPPET
+      end
+    end
+    # rubocop:disable RSpec/RepeatedExampleGroupBody
+    describe package('puppet-bolt') do
+      it { is_expected.not_to be_installed }
+    end
+
+    describe package('puppet-tools-release') do
+      it { is_expected.not_to be_installed }
+    end
+    # rubocop:enable RSpec/RepeatedExampleGroupBody
```

Comment:
```suggestion
    it { expect(package('puppet-bolt')).not_to be_installed }
    it { expect(package('puppet-tools-release')).not_to be_installed }
```

But you should make sure it's executed in the right order. You may need
```suggestion
    describe 'packages' do
      it { expect(package('puppet-bolt')).not_to be_installed }
      it { expect(package('puppet-tools-release')).not_to be_installed }
    end
```

In general the `describe resource() do` approach was recommended against when I asked about this cop.

Reply:
indeed we need the describe block, otherwise the ordering is wrong :(